app/src/main/java/com/framgia/englishconversation/widget/dialog/recordingAudio/AudioRecorder.java

```diff
@@ -0,0 +1,71 @@
+package com.framgia.englishconversation.widget.dialog.recordingAudio;
+
+import android.os.Bundle;
+import android.support.v4.app.DialogFragment;
+import android.support.v7.app.AppCompatActivity;
+
+import com.framgia.englishconversation.record.model.AudioSource;
+
+/**
+ * Created by fs-sournary.
+ * Date on 12/3/17.
+ * Description:
+ */
+
+public class AudioRecorder {
+
+    public static final String EXTRA_FILE_NAME =
+            "com.framgia.englishconversation.intent.extra.EXTRA_FILE_NAME";
+    public static final String EXTRA_FILE_PATH =
+            "com.framgia.englishconversation.intent.extra.EXTRA_FILE_PATH";
+    public static final String EXTRA_AUDIO_SOURCE =
+            "com.framgia.englishconversation.intent.extra.EXTRA_AUDIO_SOURCE";
+    public static final String NAME_RECORDING_AUDIO_FILE =
+            "RecordingAudio.3gp";
+
+    private String mFileName;
+    private String mFilePath;
+    private AudioSource mAudioSource;
+
+    private AppCompatActivity mActivity;
+    private DialogFragment mDialog;
+
+    public AudioRecorder(AppCompatActivity activity, DialogFragment dialog) {
+        mActivity = activity;
+        mDialog = dialog;
+        if (activity.getExternalCacheDir() == null) {
+            return;
+        }
+        mFileName = NAME_RECORDING_AUDIO_FILE;
+        mFilePath = activity.getExternalCacheDir().getAbsolutePath() + "/" + mFileName;
+    }
+
+    public static AudioRecorder with(AppCompatActivity activity, DialogFragment dialog) {
```

Comment:
2.8 Class member ordering

This is not the only correct solution, but it is recommended to use the following order :

Constants
Fields
Constructors
Override methods and callbacks (public or private)
Public methods
Private methods
Inner classes or interfaces